docs/documentation/namespaces-proposal.md

```diff
@@ -0,0 +1,93 @@
+# Namespaces Proposal
+
+This document contains a proposal for a guideline on how to expand the usage of 
+php namespaces to ILIAS modules and services. The suggested approach will elaborate 
+the following aspects:
+* naming conventions
+* integration of composer and autoloading
+* handling of subdirectories and interfaces
+* migration of the current structure
+* possible problems
+
+## Guideline
+
+### Naming & Structuring
+
+* The naming and structuring of all Namespaces MUST follow the PHP Standards Recommendation PSR-4: https://www.php-fig.org/psr/psr-4/.
+* The base prefix namespace is *ILIAS* and is pointing at the subdirectory */src*. Therefore, all Modules and Services MUST be located in the directory */src*. 
+* The subdirectory */classes* for Modules and Services will not be needed anymore and MUST be removed.
+* The prefixes "*class.*" and "*il*" for ILIAS classes will not be needed anymore and MUST be removed.
```

Comment:
What about the `interface` prefix? Does it make sense to remove these prefixes too?